src/overlays/actors/ovl_En_Sb/z_en_sb.c

```diff
@@ -21,53 +31,440 @@ const ActorInit En_Sb_InitVars = {
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnSb_SpawnBubbles(GlobalContext* globalCtx, EnSb* this) {
+    s32 i;
+
+    if (this->actor.unk_84 > 0) {
+        for (i = 0; i < 10; i++) {
+            func_800293E4(globalCtx, &this->actor.posRot.pos, 10.0f, 10.0f, 30.0f, 0.25f);
+        }
+    }
+}
+
+void EnSb_SetupWaitClosed(EnSb* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600004C, 1.0f, 0, SkelAnime_GetFrameCount(&D_0600004C.genericHeader), 2,
+                         0.0f);
+    this->behavior = SHELLBLADE_WAIT_CLOSED;
+    this->actionFunc = EnSb_WaitClosed;
+}
+
+void EnSb_SetupOpen(EnSb* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000194, 1.0f, 0, SkelAnime_GetFrameCount(&D_06000194.genericHeader), 2,
+                         0.0f);
+    this->behavior = SHELLBLADE_OPEN;
+    this->actionFunc = EnSb_Open;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_SHELL_MOUTH);
+}
+
+void EnSb_SetupWaitOpen(EnSb* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06002C8C, 1.0f, 0, SkelAnime_GetFrameCount(&D_06002C8C.genericHeader), 0,
+                         0.0f);
+    this->behavior = SHELLBLADE_WAIT_OPEN;
+    this->actionFunc = EnSb_WaitOpen;
+}
+
+void EnSb_SetupLunge(EnSb* this) {
+    f32 frames = SkelAnime_GetFrameCount(&D_06000124.genericHeader);
+    f32 playbackSpeed = this->actor.unk_84 > 0.0f ? 1.0f : 0.0f;
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000124, playbackSpeed, 0.0f, frames, 2, 0);
+    this->behavior = SHELLBLADE_LUNGE;
+    this->actionFunc = EnSb_Lunge;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_SHELL_MOUTH);
+}
+
+void EnSb_SetupBounce(EnSb* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060000B4, 1.0f, 0, SkelAnime_GetFrameCount(&D_060000B4.genericHeader), 2,
+                         0.0f);
+    this->behavior = SHELLBLADE_BOUNCE;
+    this->actionFunc = EnSb_Bounce;
+}
+
+void EnSb_SetupCooldown(EnSb* this, s32 changeSpeed) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_0600004C.genericHeader);
+
+    if (this->behavior != SHELLBLADE_WAIT_CLOSED) {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_0600004C.genericHeader, 1.0f, 0, frameCount, 2, 0.0f);
+    }
+    this->behavior = SHELLBLADE_WAIT_CLOSED;
+    if (changeSpeed) {
+        if (this->actor.unk_84 > 0.0f) {
+            this->actor.speedXZ = -5.0f;
+            if (this->actor.velocity.y < 0.0f) {
+                this->actor.velocity.y = 2.1f;
+            }
+        } else {
+            this->actor.speedXZ = -6.0f;
+            if (this->actor.velocity.y < 0.0f) {
+                this->actor.velocity.y = 1.4f;
+            }
+        }
+    }
+    this->timer = 60;
+    this->actionFunc = EnSb_Cooldown;
+}
+
+void EnSb_WaitClosed(EnSb* this, GlobalContext* globalCtx) {
+    // always face toward link
+    Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.rotTowardsLinkY, 10, 2000, 0);
```

Comment:
Personally I prefer hex for angles since 90 degrees is 0x4000, but the decimal numbers look nice here.

Reply:
yeah im always conflicted about this. i agree about hex for angles, but these values are very commonly nice round numbers in decimal. and its the max and min step which isnt really the angle itself, though i guess its still being used in the context of an angle value

if I should be using hex for these ill switch to that